src/backend/distributed/shared_library_init.c

```diff
@@ -664,3 +686,33 @@ NormalizeWorkerListPath(void)
 					PGC_S_OVERRIDE);
 	free(absoluteFileName);
 }
+
+
+static bool
+ReplicationFactorCheckHook(int *newval, void **extra, GucSource source)
+{
+	if (*newval != 1 && ReplicationModel == REPLICATION_MODEL_STREAMING)
+	{
+		GUC_check_errdetail("Cannot set replication factor above one when "
+							"\"citus.replication_model\" is \"streaming\".");
+
+		return false;
+	}
+
+	return true;
+}
+
+
+static bool
+ReplicationModelCheckHook(int *newval, void **extra, GucSource source)
+{
+	if (*newval == REPLICATION_MODEL_STREAMING && ShardReplicationFactor != 1)
+	{
+		GUC_check_errdetail("Cannot set replication model to \"streaming\" when "
+							"\"citus.shard_replication_factor\" greater than one.");
+
+		return false;
```

Comment:
Such interdependent checks are a bad idea, because they're order dependant.  Depending on which value is checked first, you'll get different results.  I don't think it's a good idea to have that hook here :(.  Even worse, you're supposed to be able to reliably detect whether a configuration is valid without setting things (that's where the parameters not to set values come in), and that can't work with such interdependent checks.